### DISCUSSION TITLE
Mqtt discovery failed

### DISCUSSION THREAD

**Discussion by samoswall** [HASwitchPlate/openHASP#388] [Q&A] [answered]
I use Home Assistant 2022.10.1
Integration mqtt works and sees other devices
OpenHASP works and transmits to the broker:
Topic **hasp/discovery**
{
  "node": "MyPanel",
  "mdl": "esp32-3248s035c_4MB",
  "mf": "openHASP",
  "hwid": "5443b243771c",
  "pages": 12,
  "sw": "0.6.4-dev",
  "uri": "http://192.168.1.109",
  "input": {},
  "power": [],
  "light": [],
  "dim": []
}

In the log Home Assistant **error:**
Logger: custom_components.openhasp.config_flow
Source: custom_components/openhasp/config_flow.py:75
Integration: openHASP ([documentation](https://haswitchplate.github.io/openHASP-docs/), [issues](https://github.com/HASwitchPlate/openHASP-custom-component/issues))
First occurred: 00:36:28 (2 occurrences)
Last logged: 00:37:00

Discovery Only

How to fix this error?

**Comment by fvanroie** [comment]
> In the log Home Assistant **error:**
> Logger: custom_components.openhasp.config_flow
> Source: custom_components/openhasp/config_flow.py:75

This seems to point to this [line in the openHASP Custom Component](https://github.com/HASwitchPlate/openHASP-custom-component/blob/ea2a04e049ddb59b5b0981d52560c8a902a185a6/custom_components/openhasp/config_flow.py#L75):
```python
        _LOGGER.error("Discovery Only")
```

Try changing the `.error` into `.debug`, restart HA and try again.

**Reply by CoolOn3** [reply]
Hi,
I have the same problem.
Your answer solves a problem with the Home Assistant plugin because the "Discovery Only" message should be information and not an error.
But the problem is in version 0.64-dev, because the device (file: src/mqtt/hasp_mqtt_esp.cpp line 103) is creating a sub-topic in discovery with the device id.
Example: hasp/discovery/90380cd1ade8

Changed in this update (https://github.com/HASwitchPlate/openHASP/commit/4852b6f4c4e683fd9b447836a19ae690284e7817)

HomeAssistant plugin is not seeing discovery subtopics

**Reply by dgomes** [reply]
please try the "main" branch

**Reply by CoolOn3** [reply]
![image](https://user-images.githubusercontent.com/1256245/194784772-0923f3de-f79d-4f94-947f-518dfbf4ee70.png)

Works perfectly

**Comment by fvanroie** [comment]
Two modifications have been made in the `main` branch of the [openHASP Custom Component](https://github.com/HASwitchPlate/openHASP-custom-component) that should fix this issue for openHASP firmware 0.6.4.

**Comment by samoswall** [comment, marked as answer]
Solved the problem by changing the file in Home Assistant
/config/custom_components/openhasp/manifest.json
line:                "mqtt": ["hasp/discovery"]
replaced:        "mqtt": ["hasp/discovery/#"] 
Everything works. Thanks!

**Comment by fvanroie** [comment]
So, it seems that HA supports [multiple topics for MQTT discovery](https://developers.home-assistant.io/docs/creating_integration_manifest/#mqtt). 
But that did not work. The plate get discovered again and again polluting the list of discovery devices with the same plate. 😞
